addons/mail/static/tests/discuss_app/inbox.test.js

```diff
@@ -623,3 +627,43 @@ test("emptying inbox doesn't display rainbow man in another thread", async () =>
     // weak test, no guarantee that we waited long enough for the potential rainbow man to show
     await contains(".o_reward_rainbow", { count: 0 });
 });
+
+test("Counter should update properly when a new message is added to inbox", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    const partnerId = pyEnv["res.partner"].create({ name: "Thread" });
+    const partnerUserId = pyEnv["res.partner"].create({ name: "partner1" });
+    const userId = pyEnv["res.users"].create({ partner_id: partnerUserId });
+    const messageId = pyEnv["mail.message"].create({
+        body: "not empty",
+        model: "res.partner",
+        needaction: true,
+        needaction_partner_ids: [serverState.partnerId],
+        res_id: partnerId,
+    });
+    pyEnv["mail.notification"].create([
+        {
+            mail_message_id: messageId,
+            notification_type: "inbox",
+            res_partner_id: serverState.partnerId,
+        },
+    ]);
+    const env = await start();
+    rpc = rpcWithEnv(env);
+    await openDiscuss();
+    await contains("button", { text: "Inbox", contains: [".badge", { count: "1" }] });
+    const mention = [serverState.partnerId];
+    withUser(userId, () =>
+        rpc("/mail/message/post", {
```

Comment:
If possible, a multi-tab test would be more realistic than withUser + rpc. (can just do the actual mention flow as another user rather than hard-coding message stuff)

Reply:
It's not possible to have tab with different users